Projects/05matchGame/index.js

```diff
@@ -0,0 +1,402 @@
+    await handleRequestDeckId();
+    await handleRequestCards();
+    await handleStartingGame();
+
+    //get deckId from API;
+    async function handleRequestDeckId() {
+      return fetch(
+        `https://deckofcardsapi.com/api/deck/new/shuffle/?deck_count=1`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          deckId = data.deck_id;
+        })
+        .catch(() => {
+          notResponseApi();
+        });
+    }
+
+    //get cards from API;
+    async function handleRequestCards() {
+      return fetch(
+        `https://deckofcardsapi.com/api/deck/${deckId}/draw/?count=8`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          savedCards.push(data.cards);
+          let mergeSavedCards = savedCards.flat(1);
+
+          //copy the 8 cards from API - make an array with 16 cards - 8 pairs of cards;
+          let cloneSavedCards = [...mergeSavedCards];
+
+          allFaceCards = mergeSavedCards.concat(cloneSavedCards);
+          allFaceCards
+            .map((cardFace) => {
+              allSavedCards.push(cardFace.images.png);
+
+              //randomize the cards from API;
+              return allSavedCards.sort(() => Math.random() - 0.5);
+            })
+            .join("");
+        })
+        .catch(() => {
+          notResponseApi();
+        });
+    }
+
+    //show popup with sysMsg - not response from API;
+    function notResponseApi() {
+      document.querySelector(".pageGame").innerHTML = "";
+      uiElements.$popupContainer.style.display = "block";
+      sysMsg.popupWinLoseNotResponse.innerHTML = sysMsg.notResponseAPI;
+      sysMsg.popupWinLoseNotResponse.classList.add("centerSysMsg");
+    }
+
+    //if "x" is clicked, hide the popup;
+    function closePopup() {
+      document.querySelector(".close").onclick = () => {
+        uiElements.$popupContainer.style.display = "none";
+      };
+    }
+
+    //generate the number of card slots;
+    async function handleStartingGame() {
+      uiElements.$containerCards.innerHTML = "";
+
+      generateCardSlots();
+      flipOnClick();
+      generateTable();
+      uiElements.$resetBtn.addEventListener("click", resetBtn);
+    }
+
+    //function to generate the slots with backCard and faceCard(from API);
+    function generateCardSlots() {
+      for (let i = 0; i < slots; i++) {
+        HTMLCards = `<div class="cardSlot" >
+        <div class="flipCard">
+            <div class="backCard">
+                <img src="images/cardback_158.png" alt="backCard"/>
+            </div>
+            <div class="faceCard">
+              <img src="${allSavedCards[i]}"/>
+            </div>
+        </div>`;
+
+        uiElements.$containerCards.innerHTML += HTMLCards;
+      }
+    }
+
+    //function to add and remove class on click;
+    function flipOnClick() {
+      let flipCards = document.querySelectorAll(".flipCard");
+      for (let i = 0; i < flipCards.length; i++) {
```

Comment:
the name of the class it's not flipCards, should be simple like 'cards' OR 'card'

Reply:
@kvmladenov done